web/src/hooks/usePoiFeatures.ts

```diff
@@ -0,0 +1,77 @@
+import { useCallback, useEffect, useMemo } from 'react'
+import { useNavigate, useSearchParams } from 'react-router-dom'
+
+import { GeoJsonPoi, isMultipoi, PoiFeature, PoiModel } from 'api-client'
+
+const multipoiKey = 'multipoi'
+
+const usePoiFeatures = (
+  features: PoiFeature[],
+  pois: PoiModel[],
+  slug?: string
+): {
+  selectFeatureOnMap: (newFeatureOnMap: PoiFeature | null) => void
+  selectPoiFeatureInList: (newPoiFeature: GeoJsonPoi | null) => void
+  currentFeatureOnMap: PoiFeature | null
+  currentPoi: PoiModel | null
+  poiListFeatures: GeoJsonPoi[]
+} => {
+  const poiFeatures = useMemo(() => features.flatMap(feature => feature.properties.pois), [features])
+  const currentPoi = useMemo(() => pois.find(poi => slug === poi.slug) ?? null, [pois, slug])
+  const [searchParams, setSearchParams] = useSearchParams()
+  const navigate = useNavigate()
+
+  const currentFeatureOnMap =
+    useMemo(
+      () =>
+        searchParams.has(multipoiKey)
+          ? features.find(feature => feature.id === searchParams.get(multipoiKey))
+          : features.find(feature => feature.properties.pois.some(poi => poi.slug === slug)),
+      [features, searchParams, slug]
+    ) ?? null
+
+  useEffect(() => {
+    if (currentPoi && !currentFeatureOnMap?.properties.pois.some(poiFeature => poiFeature.slug === currentPoi.slug)) {
+      // remove multipoi from search params if current poi is not part of this multipoi
+      setSearchParams(new URLSearchParams())
+    }
+  }, [currentFeatureOnMap?.properties.pois, currentPoi, setSearchParams])
```

Comment:
So I tested this again by uncommenting and I did not find a case where this is necessary, everything worked just fine without. Do you have any more information on how to reproduce this edge case for which this is necessary?

Reply:
So I think I added this to keep a consistent state and not display sth. weird. Try navigating to `http://localhost:9000/testumgebung/en/locations/rathausplatz?multipoi=4`. Imo this url should not be valid. Therefore I added this useEffect.

Reply:
hm for me also everything works fine @sarahsporck 
I go to the url and the camera goes to the multipoi. I can't see any difference on chrome nor firefox with or without that. I don't think this is needed as a hook

Reply:
So when you go to the given url, without this `useEffect` you should see that the multipoi is selected on the map and the camera also moves to the multipoi, but in the list `Rathausplatz` is selected. Please tell me I don't go crazy here ^^

Reply:
you are right @sarahsporck :)